pkg/controller/manage/kmesh_manage.go

```diff
@@ -209,9 +209,25 @@ func isPodBeingDeleted(pod *corev1.Pod) bool {
 	return pod.ObjectMeta.DeletionTimestamp != nil
 }
 
-func shouldEnroll(pod *corev1.Pod) bool {
-	mode := pod.Labels[constants.DataPlaneModeLabel]
-	return strings.EqualFold(mode, constants.DataPlaneModeKmesh)
+func shouldEnroll(client kubernetes.Interface, pod *corev1.Pod) bool {
+	// Check if the Pod's label indicates it should be managed by Kmesh
+	if strings.EqualFold(pod.Labels[constants.DataPlaneModeLabel], constants.DataPlaneModeKmesh) {
+		return true
+	}
+	if pod.Annotations[KmeshRedirectionAnnotation] == "enabled" {
```

Comment:
nit：It is recommended that `KmeshRedirectionAnnotation` be placed in constants.go.